includes/class-freemius.php

```diff
@@ -7952,17 +8004,186 @@ private function maybe_network_activate_addon_license( $license = null ) {
+                                $installs_by_blog_map[ $blog_id ] = self::get_all_sites( $fs->get_module_type(), $blog_id );
+                            }
+
+                            $installs = $installs_by_blog_map[ $blog_id ];
+                            $install  = null;
+
+                            if ( isset( $installs[ $fs->get_slug() ] ) ) {
+                                $install = $installs[ $fs->get_slug() ];
+
+                                if ( is_object( $install ) &&
+                                    is_numeric( $install->id ) &&
+                                    is_numeric( $install->user_id ) &&
+                                    FS_Plugin_Plan::is_valid_id( $install->plan_id )
+                                ) {
+                                    $install = clone $install;
+                                } else {
+                                    $install = null;
+                                }
+                            }
+
+                            if ( is_object( $install ) && FS_Plugin_License::is_valid_id( $install->license_id ) ) {
+                                $has_install_with_license = true;
+                                break;
+                            }
+
+                            if ( ! $fs->is_site_delegated_connection( $blog_id ) && ! is_object( $install ) ) {
+                                if ( ! isset( $site_info_by_blog_map[ $blog_id ] ) ) {
+                                    $site_info_by_blog_map[ $blog_id ] = $fs->get_site_info( $site );
+                                }
+
+                                $product_sites[] = $site_info_by_blog_map[ $blog_id ];
+                            }
+                        }
+
+                        if ( $has_install_with_license ) {
+                            // Do not try to activate the license in the network level if there's any install with a license.
+                            continue;
+                        }
+                    }
+                }
+
+                $fs->activate_migrated_license(
+                    $license->secret_key,
+                    null,
+                    null,
+                    $sites,
+                    ( $current_blog_id > 0 ? $current_blog_id : null )
+                );
+            }
+        }
+
         /**
          * @author Leo Fajardo (@leorw)
          * @since 2.3.0
          *
+         * @param string|null $license_key
+         * @param bool        $flush
+         *
          * @return FS_Plugin_License
          */
-        private function get_addon_active_parent_license() {
+        function get_active_parent_license( $license_key = null, $flush = true ) {
             $parent_licenses_endpoint = "/plugins/{$this->get_id()}/parent_licenses.json?filter=activatable";
-            $parent_instance          = $this->get_parent_instance();
+            $parent_instance          = $this->is_addon() ?

```

Comment:
@fajardoleo I would suggest to simplify the logic that calculates `$fs` as following:
```
$fs = $this;
if ( $this->is_addon() ) {
    $parent_instance = $this->get_parent_instance();

    if ( is_object( $parent_instance ) && $parent_instance->is_registered() ) {
        $fs = $parent_instance;
    }
}
```